zygrader/config/versioning.py

```diff
@@ -97,6 +97,9 @@ def versioning_update_preferences():
     if compare_versions(version, user_version):
         config["spooky_mode"] = False
 
+    # Code here will always be run until the next version bump
+    config["output_dir"] = "~"
+
```

Comment:
Also, this could make use of `preferences.DEFAULT_PREFERENCES`. Perhaps there should be an unconditional check for non-existent preferences that sets them to the defaults.

Reply:
For development purposes it is nice to have an "always run this" section in versioning. Without it, you can run into versioning issues:
1. Feature A is added and bumps the version (say to 5.0.1) because a new default needs to be set
2. Time passes (zygrader is run)
3. Feature B is added and also needs versioning, but because the config version number is 5.0.1 the versioning code never runs.

So it's helpful to have a section like this that is always run until that version is officially released. With that release we move it to a conditional.

---

However, in this case we should probably make an exception for adding new preferences with a simple loop in the default preferences.

Reply:
Ah, makes sense.